src/vs/workbench/api/browser/mainThreadChatCodeMapper.ts

```diff
@@ -56,9 +59,35 @@ export class MainThreadChatCodemapper extends Disposable implements MainThreadCo
 	$handleProgress(requestId: string, data: ICodeMapperProgressDto): Promise<void> {
 		const response = this._responseMap.get(requestId);
 		if (response) {
+			const edits = data.edits;
 			const resource = URI.revive(data.uri);
-			response.textEdit(resource, data.edits);
+			if (!edits.length) {
+				response.textEdit(resource, []);
+			} else if (areTextEdits(edits)) {
+				response.textEdit(resource, edits);
+			} else {
+				const cellEdits: ICellEditOperation[] = [];
+				edits.forEach(dto => {
+					if (dto.editType === CellEditType.Replace) {
+						cellEdits.push({
+							editType: dto.editType,
+							index: dto.index,
+							count: dto.count,
+							cells: dto.cells.map(NotebookDto.fromNotebookCellDataDto)
+						});
+					}
+				});
+				response.notebookEdit(resource, cellEdits);
```

Comment:
Is this a better place for `chatService.ts/NotebookEdits#toCellEditReplaceOperation`

Reply:
Sorry, but I'm not following you here

Reply:
Also I ended up cleaning a few of the types as part of the commit. Hence `ICellEditReplaceOperation` is no longer available.
I fixed the types where I was filtering to keep just Replace edit operations, now we support all other operations that `NotebookEdit` supports (replace cell and notebook metadata, insert/replace/delete cells)